subprojects/model-core/src/main/java/org/gradle/internal/instantiation/generator/AsmBackedClassGenerator.java

```diff
@@ -1770,7 +1769,7 @@ private ReturnTypeEntry(String fieldName, String getterName) {
     private static String getBuildServiceName(PropertyMetadata property) {
         ServiceReference annotation = property.findAnnotation(ServiceReference.class);
         if (annotation != null) {
-            return StringUtils.trimToNull(annotation.value());
+            return annotation.value();
```

Comment:
As per feedback from @lptr

Reply:
We still do, but when interpreting the service name at runtime, like [here](https://github.com/gradle/gradle/pull/22987/files#diff-cea8ba74f470c721fcb0c610255af0b2946bb5122597f00d21e0a58232ba4837R101).